Matrix_On_Air/matrix_on_air.py

```diff
@@ -0,0 +1,239 @@
+from adafruit_display_shapes.rect import Rect
+from adafruit_bitmap_font import bitmap_font
+from adafruit_matrixportal.network import Network
+from adafruit_matrixportal.matrix import Matrix
+
+# Get wifi details and more from a secrets.py file
+try:
+    from secrets import secrets
+except ImportError:
+    print("WiFi secrets are kept in secrets.py, please add them there!")
+    raise
+
+# Set up where we'll be fetching data from
+# Adafruit YouTube channel:
+CHANNEL_ID = (
+    "UCpOlOeQjj7EsVnDh3zuCgsA"  # this isn't a secret but you have to look it up
+)
+
+DATA_SOURCE = (
+    "https://www.googleapis.com/youtube/v3/search?part=snippet&channelId="
+    + CHANNEL_ID
+    + "&type=video&eventType=live&key="
+    + secrets["youtube_token"]
+)
+DATA_LOCATION1 = ["items"]
+
+# Number of seconds between checking, if this is too quick the query quota will run out
+UPDATE_DELAY = 300
+
+# Times are in 24-hour format for simplification
+OPERATING_TIME_START = "12:00"  # what hour to start checking
+OPERATING_TIME_END = "19:00"  # what hour to stop checking
+
+# --- Display setup ---
+matrix = Matrix()
+display = matrix.display
+network = Network(status_neopixel=board.NEOPIXEL, debug=False)
+
+# --- Drawing setup ---
+# Create a Group
+group = displayio.Group(max_size=22)
+# Create a bitmap object
+bitmap = displayio.Bitmap(64, 32, 2)  # width, height, bit depth
+# Create a color palette
+color = displayio.Palette(4)
+color[0] = 0x000000  # black
+color[1] = 0xFF0000  # red
+color[2] = 0x444444  # dim white
+color[3] = 0xDD8000  # gold
+# Create a TileGrid using the Bitmap and Palette
+tile_grid = displayio.TileGrid(bitmap, pixel_shader=color)
+# Add the TileGrid to the Group
+group.append(tile_grid)
+
+# draw the frame for startup
+rect1 = Rect(0, 0, 2, 32, fill=color[2])
+rect2 = Rect(62, 0, 2, 32, fill=color[2])
+rect3 = Rect(2, 0, 9, 2, fill=color[2])
+rect4 = Rect(53, 0, 9, 2, fill=color[2])
+rect5 = Rect(2, 30, 12, 2, fill=color[0])
+rect6 = Rect(50, 30, 12, 2, fill=color[0])
+
+group.append(rect1)
+group.append(rect2)
+group.append(rect3)
+group.append(rect4)
+group.append(rect5)
+group.append(rect6)
+
+
+def redraw_frame():  # to adjust spacing at bottom later
+    rect5.fill = color[2]
+    rect6.fill = color[2]
+
+
+# draw the wings w line shapes
+wing1 = Line(3, 3, 10, 3, color[1])  # x-start, y-start, x-end, y-end, color
+wing2 = Line(4, 4, 10, 4, color[1])
+wing3 = Line(5, 6, 10, 6, color[1])
+wing4 = Line(6, 7, 10, 7, color[1])
+wing5 = Line(7, 9, 10, 9, color[1])
+wing6 = Line(8, 10, 10, 10, color[1])
+wing7 = Line(54, 3, 61, 3, color[1])
+wing8 = Line(54, 4, 60, 4, color[1])
+wing9 = Line(54, 6, 59, 6, color[1])
+wing10 = Line(54, 7, 58, 7, color[1])
+wing11 = Line(54, 9, 57, 9, color[1])
+wing12 = Line(54, 10, 56, 10, color[1])
+
+wing_lines = [
+    wing1,
+    wing2,
+    wing3,
+    wing4,
+    wing5,
+    wing6,
+    wing7,
+    wing8,
+    wing9,
+    wing10,
+    wing11,
+    wing12,
+]
```

Comment:
This could probably be reduced significantly with:
```
# draw the wings w line shapes
wing_lines = []

wing_lines.append(Line(3, 3, 10, 3, color[1]))  # x-start, y-start, x-end, y-end, color
wing_lines.append(Line(4, 4, 10, 4, color[1]))
wing_lines.append(Line(5, 6, 10, 6, color[1]))
wing_lines.append(Line(6, 7, 10, 7, color[1]))
wing_lines.append(Line(7, 9, 10, 9, color[1]))
wing_lines.append(Line(8, 10, 10, 10, color[1]))
wing_lines.append(Line(54, 3, 61, 3, color[1]))
wing_lines.append(Line(54, 4, 60, 4, color[1]))
wing_lines.append(Line(54, 6, 59, 6, color[1]))
wing_lines.append(Line(54, 7, 58, 7, color[1]))
wing_lines.append(Line(54, 9, 57, 9, color[1]))
wing_lines.append(Line(54, 10, 56, 10, color[1]))
```

Reply:
great, I just incorporated that, thanks!